myapp/src/pages/WelcomePage.js

```diff
@@ -0,0 +1,16 @@
+import { useLocation} from "react-router-dom";
+
+function WelcomePage() {
+  const location = useLocation();
+
+  const data = location.state ?? {};
+  return (
+    <div className="flex items-center justify-center w-full h-screen bg-violet-100">
+      <h1 className="text-5xl font-semibold text-violet-600">
+        Welcome {data.username ?? "null"}!
```

Comment:
Why are displaying `Welcome null` when a username is not there. I think we should show a generic text like `Welcome` or `Welcome user`, or not show anything at all.

Reply:
Change it.